File: leetcode/js/maximum-odd-binary-number.js
/**
 * @param {string} s
 * @return {string}
 */
var maximumOddBinaryNumber = function (s) {
  let countOne = 0;
  let countZero = 0;

  for (const b of s) {
    if (b === "1") {
      countOne++;
    } else {
      countZero++;
    }
  }

  return "1".repeat(countOne - 1) + "0".repeat(countZero) + "1";
};

File: leetcode/rust/maximum-odd-binary-number.rs
impl Solution {
    pub fn maximum_odd_binary_number(s: String) -> String {
        let mut count_one = 0;
        let mut count_zero = 0;

        for b in s.chars() {
            if b == '1' {
                count_one += 1;
            } else {
                count_zero += 1;
            }
        }

        let str_one = "1".repeat(count_one - 1);
        let str_zero = "0".repeat(count_zero);

        format!("{str_one}{str_zero}1")
    }
}
